lib/puppet/provider/service/systemd.rb

```diff
@@ -12,7 +12,7 @@
   def self.instances
     i = []
     output = systemctl('list-unit-files', '--type', 'service', '--full', '--all',  '--no-pager')
-    output.scan(/^(\S+)\s+(disabled|enabled)\s*$/i).each do |m|
+    output.scan(/^(\S+)\s+(disabled|enabled|masked)\s*$/i).each do |m|
```

Comment:
The list to scan should include `manual` per the docs in `puppet describe service`:
```
$ puppet describe service
...
- **enable**
    Whether a service should be enabled to start at boot.
    This property behaves quite differently depending on the platform;
    wherever possible, it relies on local tools to enable or disable
    a given service.
    Valid values are `true`, `false`, `manual`, `mask`.
    Requires features enableable.
...
```